internal/testrunner/runners/system/system.go

```diff
@@ -2,30 +2,134 @@ package system
 
 import (
 	"fmt"
+	"os"
+	"time"
 
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/common"
+	"github.com/elastic/elastic-package/internal/kibana/ingestmanager"
+	"github.com/elastic/elastic-package/internal/logger"
 	"github.com/elastic/elastic-package/internal/testrunner"
+	"github.com/elastic/elastic-package/internal/testrunner/runners/system/servicedeployer"
 )
 
+func init() {
+	testrunner.RegisterRunner(TestType, Run)
+}
+
 const (
 	// TestType defining system tests
 	TestType testrunner.TestType = "system"
 )
 
 type runner struct {
-	testFolderPath string
+	testFolder      testrunner.TestFolder
+	packageRootPath string
+	stackSettings   stackSettings
+}
+
+type stackSettings struct {
+	elasticsearch struct {
+		host     string
+		username string
+		password string
+	}
+	kibana struct {
+		host string
+	}
 }
 
 // Run runs the system tests defined under the given folder
-func Run(testFolderPath string) error {
-	r := runner{testFolderPath}
+func Run(testFolder testrunner.TestFolder, packageRootPath string) error {
+	r := runner{
+		testFolder,
+		packageRootPath,
+		getStackSettingsFromEnv(),
+	}
 	return r.run()
 }
 
 func (r *runner) run() error {
-	fmt.Println("system run", r.testFolderPath)
+	// Step 1. Setup service.
+	// Step 1a. (Deferred) Tear down service.
+	logger.Info("Setting up service...")
+	serviceRunner, err := servicedeployer.Factory(r.packageRootPath)
+	if err != nil {
+		return errors.Wrap(err, "could not create service runner")
+	}
+
+	ctxt := common.MapStr{}
```

Comment:
I think it's still early and there is no implementation using `MapStr`. Do you plan to use the MapStr to flatten JSON object representation? If so, I wonder if standars `encoding/json` is sufficient.

Reply:
Also, this is the reason why I don't flatten the expected results in the pipeline tests. Maybe I should introduce it there.

Reply:
I'm not 100% sure yet how this usage is going to evolve. My current thinking is to use `ctxt` to pass some arbitrary key-value pair data between the various steps of testing but this is something I'm experimenting with.

For test results, since those are consistently going to be present for each test run, I probably wouldn't use `ctxt` but something else. I haven't gotten that far yet so I'm not sure yet.